build_tools/build_detect_platform

```diff
@@ -454,17 +454,22 @@ if test -z "$PORTABLE"; then
   elif [ "$TARGET_OS" != AIX ] && [ "$TARGET_OS" != SunOS ]; then
     COMMON_FLAGS="$COMMON_FLAGS -march=native "
   elif test "$USE_SSE"; then
-    COMMON_FLAGS="$COMMON_FLAGS -msse4.2"
+    COMMON_FLAGS="$COMMON_FLAGS -msse4.2 -mpclmul"
   fi
 elif test "$USE_SSE"; then
-  COMMON_FLAGS="$COMMON_FLAGS -msse4.2"
+  COMMON_FLAGS="$COMMON_FLAGS -msse4.2 -mpclmul"
 fi
 
 $CXX $PLATFORM_CXXFLAGS $COMMON_FLAGS -x c++ - -o /dev/null 2>/dev/null <<EOF
   #include <cstdint>
   #include <nmmintrin.h>
+  #include <wmmintrin.h>
   int main() {
     volatile uint32_t x = _mm_crc32_u32(0, 0);
+    const auto a = _mm_set_epi64x(0, 0);
+    const auto b = _mm_set_epi64x(0, 0);
+    const auto c = _mm_clmulepi64_si128(a, b, 0x00);
+    auto d = _mm_cvtsi128_si64(c);
```

Comment:
I'm not an expert on this but is there any case where _mm_set_epi64x() is not supported but _mm_crc32_u32() is supported? In that case, we could have use fast CRC but ends up use slow CRC.

Reply:
I don't know how to find if such a compiler version exists. But if USE_SSE=1 is set and this fails, now we're gonna fail the build.

Reply:
In my understanding _mm_set_epi64x was supported at least since SSE2(Pentium 4 in 2001) and should be supported if _mm_crc32_u32(supported since SSE4.2 on Nehalem in 2008?) is supported. _mm_clmulepi64_si128 was supported later on Westmere in 2010. But compiling wise, I don't know if there's compiler that can compile CRC32 intrinsics but not PCLMUL ones since there has been many years passed and it's unlikely people are using something before 2010. But I have separated the compile tests anyways and used HAVE_SSE42 and HAVE_PCLMUL tags.